sisl/grid.py

```diff
@@ -179,6 +179,84 @@ def interp(self, shape, order=1, mode="wrap", **kwargs):
 
         # Apply the scipy.ndimage.zoom function and return a new grid
         return self.apply(ndimage_zoom, zoom_factors, mode=mode, order=order, **kwargs)
+    
+    def transform_axes(self, axes=np.eye(3), output_shape=None, mode="constant", order=1, **kwargs):
```

Comment:
Instead of having `axes` as input, perhaps the first argument could be either:
- `SuperCell` input
- ndarray, passed to `SuperCell` to generate sc
- `Grid`, to get parent sc and `output_shape`

I.e. something like this:
```python
def ...(self, obj, *args, ...)
    def_shape = self.shape
    if isinstance(obj, Grid):
        sc = obj.sc
        def_shape = obj.shape
    elif not isinstance(obj, SuperCell):
        sc  = obj
    else:
        sc = SuperCell(obj)
    ... do stuff with sc instead of axes
```

While I can see you re-use the `output_shape` keyword from `scipy`, I am not sure this is the best wording... I.e. all other places we use `shape`... Hmm.. really confusing `scipy` choose this wording...

Reply:
Yes, good idea. 

However, do I keep the feature that the vectors in `sc` only indicate the directions and are scaled as needed? In the case of cartesian coordinates it seems much more reasonable, but I don't know about the other cases.